File: day1/1.py
import fileinput

def getFloor(directionsFile):
    currentFloor = 0
    for line in directionsFile:
        for floor in line:
            if floor == '(':
                currentFloor += 1
            elif floor == ')':
                currentFloor -= 1

    return currentFloor

print getFloor(fileinput.input())

File: day1/1.js
'use strict';
const fs = require('fs');

let fileName = process.argv[2] || 'input.txt';
fs.readFile(fileName, (err, data) => {
  console.log(getFloor(String(data)));
});

function getFloor(directions) {
  return directions.split('').reduce((previousFloor, floor) => {
    if (floor === '(') {
      return previousFloor + 1;
    } else if (floor === ')') {
      return previousFloor - 1;
    }
    return previousFloor;
  }, 0);
}
